src/lib/phdi-building-blocks/phdi_building_blocks/schemas.py

```diff
@@ -0,0 +1,251 @@
+            )
+            if values_from_resource != {}:
+                raw_schema_data.append(values_from_resource)
+
+        # Write data to parquet
+        writer = write_schema_table(
+            raw_schema_data, output_file_name, output_format, writer
+        )
+
+        # Check for an additional page of query results.
+        for link in query_result.get("link"):
+            if link.get("relation") == "next":
+                next_page_url = link.get("url")
+                response = query_fhir_server(
+                    credential_manager, specific_url=next_page_url
+                )
+                break
+            else:
+                response = None
+
+        if response is None:
+            additional_page = False
+    if writer is not None:
+        writer.close()
+
+
+def generate_schema(
+    fhir_url: str,
+    schema_path: pathlib.PosixPath,
+    output_path: pathlib.PosixPath,
+    output_format: Literal["parquet"],
+):
+    """
+    Given the url for a FHIR server, the location of a schema file, and and output
+    directory generate the specified schema and store the tables in the desired
+    location.
+
+    :param fhir_url: URL to a FHIR server.
+    :schema_path: A path to the location of a YAML schema config file.
+    :output_path: A path to the directory where tables of the schema should be written.
+    :output_format: Specifies the file format of the tables to be generated.
+    """
+
+    schema = load_schema(schema_path)
+    schema_name = list(schema.keys())[0]
+    schema = schema[schema_name]
+    output_path = pathlib.Path(schema_name)
+
+    credential_manager = AzureFhirserverCredentialManager(fhir_url)
+
+    for resource_type in schema.keys():
+        make_resource_type_table(
+            resource_type, schema, output_path, output_format, credential_manager
+        )
+
+
+def write_schema_table(
+    data: List[dict],
+    output_file_name: pathlib.PosixPath,
+    file_format: Literal["parquet"],
+    writer: pq.ParquetWriter = None,
+):
+    """
+    Write schema data to a file given the data, a path to the file including the file
+    name, and the file format.
+
+    :param data: A list of dictionaries specifying the data for each row of a table
+    where the keys of each dict correspond the columns and the values contain the data
+    for each entry in a row.
+    :param output_file_name: Full name for the file where the table is to be written.
+    :output_format: Specifies the file format of the table to be written.
+    :param writer: A writer object that can be kept open between calls of this function
+    to support file formats that cannot be appended to after being written
+    (e.g. parquet).
+    """
+
+    if file_format == "parquet":
+        table = pa.Table.from_pylist(data)
+        if writer is None:
+            writer = pq.ParquetWriter(output_file_name, table.schema)
+        writer.write_table(table=table)
+        return writer
+
+
+def get_schema_summary(
```

Comment:
This should be called something like `get_metadata` or `get_summary_statistics` or `preview_tables`.